core/src/main/java/io/questdb/cairo/mv/MatViewDefinition.java

```diff
@@ -492,9 +518,20 @@ private static void readDefinitionBlock(
                     .put(matViewToken.getTableName())
                     .put(']');
         }
+        int baseTableTimestampType = 0;
```

Comment:
this is more involved that this.


TLDR: we have to write timestamp type into view definition file to make sure materialized view and the source SQL/table are permanently aligned on timestamp type. E.g. they must never diverge.

view definition file is designed to be easily upgradable. E.g. it contains blocks of data with block types. Old versions are designed to skip block types they do not understand. We should write a new block (new block type) to the view definition file, which will contain timestamp type. Older version will not read it and assume "micros" but newer version will read this value from the file instead of getting it from the engine against what essentially is a dynamic table (it can be dropped and re-created)